sdv/tabular/base.py

```diff
@@ -322,21 +333,42 @@ def _sample_batch(self, num_rows=None, max_tries=100, batch_size_per_try=None,
         if not batch_size_per_try:
             batch_size_per_try = num_rows * 10
 
-        sampled, num_valid = self._sample_rows(
-            num_rows, conditions, transformed_conditions, float_rtol)
+        if not progress_bar:
+            progress_bar = tqdm(total=num_rows)
+
+        counter = 0
+        num_valid = 0
+        prev_num_valid = None
+        remaining = num_rows
+        sampled = pd.DataFrame()
 
-        counter = 1
         while num_valid < num_rows:
             if counter >= max_tries:
                 break
 
-            remaining = num_rows - num_valid
-
-            LOGGER.info(f'{remaining} valid rows remaining. Resampling {batch_size_per_try} rows')
+            prev_num_valid = num_valid
             sampled, num_valid = self._sample_rows(
                 batch_size_per_try, conditions, transformed_conditions, float_rtol, sampled,
             )
 
+            num_increase = min(num_valid - prev_num_valid, remaining)
```

Comment:
In what case would `remaining` be less than `num_valid - prev_num_valid`

Reply:
Since we are sampling up to `batch_size_per_try` rows, we could sample more than the number of remaining rows.